llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -5518,6 +5517,9 @@ InstructionCost LoopVectorizationCostModel::computePredInstDiscount(
     // includes the scalarization overhead of the predicated instruction.
     InstructionCost VectorCost = getInstructionCost(I, VF);
 
+    if (VectorCost == InstructionCost::getInvalid())
+      continue;
```

Comment:
Should this return here? The comment for the loop says it computes the discount of scalarizing the whole expression, so skipping here means we just ignored some costs, while it is not possible to scalarize the whole expression.